src/main/java/uk/ac/ebi/eva/utils/FileWithHeaderNumberOfLinesEstimator.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.ac.ebi.eva.pipeline.listeners.StepProgressListener;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Estimate the number of lines in a file.
+ */
+public class FileWithHeaderNumberOfLinesEstimator {
+    private static final Logger logger = LoggerFactory.getLogger(FileWithHeaderNumberOfLinesEstimator.class);
+
+    private static final int NUMBER_OF_LINES = 150;
```

Comment:
After some discussion, we agreed to increase this number to 10K to get more precision, but then, if the file is shorter than that, it is reasonable to output the actual line number, as we know that at some point